src/plugins/intel_cpu/src/nodes/rope.cpp

```diff
@@ -7,17 +7,281 @@
+        } else {
+            auto half_rotary_ndims = jcp_.rotary_ndims / 2;
+            size_t steps = 0;
+            for (size_t i = 0; i < half_rotary_ndims / vec_size; i++) {
+                rotary_half(vec_size);
+                steps += vec_size;
+            }
+            if (half_rotary_ndims % vec_size != 0) {
+                rotary_half(half_rotary_ndims % vec_size);
+                steps += half_rotary_ndims % vec_size;
+            }
+        }
+        this->postamble();
+        for (const auto& emitter : emitters) {
+            if (emitter.second)
+                emitter.second->emit_data();
+        }
+    }
+    void rotary_half(size_t step) {
+        // for (; i < half_rotary_dims; i++) {
+        //     auto src0 = src[i];
+        //     auto src1 = src[i + half_rotary_dims];
+        //     dst[i] = cos[i] * src0 - sin[i] * src1;
+        //     dst[i + half_rotary_dims] = cos[i + half_rotary_dims] * src1 + sin[i + half_rotary_dims] * src0;
+        // }
+        auto half_rotary_ndims = jcp_.rotary_ndims / 2;
+        // src0: src[i]
+        load(vmm_src0, reg_src, jcp_.src_prc, step, false);
+        // src1: src[i + halfRotaryNdims]
+        lea(reg_tmp, ptr[reg_src + half_rotary_ndims * jcp_.src_prc.size()]);
+        load(vmm_src1, reg_tmp, jcp_.src_prc, step, false);
+        // cos[i]
+        load(vmm_cos, reg_cos, ov::element::f32, step, false);
+        // sin[i]
+        load(vmm_sin, reg_sin, ov::element::f32, step, false);
+        // sin[i] * src1
+        uni_vmulps(vmm_dst0, vmm_sin, vmm_src1);
+        // cos[i] * src0 - sin[i] * src1
+        vfmsub231ps(vmm_dst0, vmm_cos, vmm_src0);
+        store(reg_dst, vmm_dst0, jcp_.dst_prc, step);
+
+        // cos[i + halfRotaryNdims]
+        lea(reg_tmp, ptr[reg_cos + half_rotary_ndims * sizeof(float)]);
+        load(vmm_cos, reg_tmp, ov::element::f32, step, false);
+        // sin[i + halfRotaryNdims]
+        lea(reg_tmp, ptr[reg_sin + half_rotary_ndims * sizeof(float)]);
+        load(vmm_sin, reg_tmp, ov::element::f32, step, false);
+        // cos[i + half_rotary_dims] * src1
+        uni_vmulps(vmm_dst0, vmm_cos, vmm_src1);
+        // cos[i + half_rotary_dims] * src1 + sin[i + half_rotary_dims] * src0
+        vfmadd231ps(vmm_dst0, vmm_sin, vmm_src0);
+        lea(reg_tmp, ptr[reg_dst + half_rotary_ndims * jcp_.dst_prc.size()]);
+        store(reg_tmp, vmm_dst0, jcp_.dst_prc, step);
+
+        add(reg_src, jcp_.src_prc.size() * step);
+        add(reg_dst, jcp_.dst_prc.size() * step);
+        add(reg_cos, sizeof(float) * step);
+        add(reg_sin, sizeof(float) * step);
+    }
+    void rotary_interleave(size_t step) {
+        // for (size_t j = 0; i < rotary_dims; i += 2, j++) {
+        //     dst[i] = cos[j] * x[i] - sin[j] * x[i + 1];
+        //     dst[i + 1] = cos[j] * x[i + 1] + sin[j] * x[i];
+        // }
+        load(vmm_src0, reg_src, jcp_.src_prc, step, false);
+        lea(reg_tmp, ptr[reg_src + step * jcp_.src_prc.size()]);
```

Comment:
Load emitter takes constant offset as second input param: https://github.com/openvinotoolkit/openvino/blob/fc0b82c1be18f793b41e894981a061b90c31a1e1/src/plugins/intel_cpu/src/emitters/plugin/x64/jit_load_store_emitters.cpp#L105. It means you can avoid multiple lea instruction by passing {reg_src, step * jcp_.src_prc.size()} to emit_code().

Reply:
Done.